contrib/win32/win32compat/w32fd.c

```diff
@@ -1105,7 +1158,29 @@ spawn_child_internal(const char* cmd, char *const argv[], HANDLE in, HANDLE out,
 	do {
 		if (as_user) {
 			debug3("spawning %ls as user", t);
-			b = CreateProcessAsUserW(as_user, NULL, t, NULL, NULL, TRUE, flags, NULL, NULL, &si, &pi);
+			LPVOID lpEnvironment = NULL;
+			BOOL foreign_user = FALSE; /* By default, we assume that the user environment block does not need to be loaded. */
+			wchar_t* as_user_name = get_username_from_token(as_user);
+			if (as_user_name) {
+				if (wcsncmp(L"sshd", as_user_name, sizeof("sshd") - 1) != 0) { /* Ignore any names starting with `sshd` (this is the service name). */
+					wchar_t* current_name = get_username_from_environment();
+					if (current_name) {
+						foreign_user = wcscmp(as_user_name, current_name) != 0;
```

Comment:
I don't think it makes sense to check the USERNAME environment variable when constructing the user environment block, since this check is always true.